awx/main/tasks.py

```diff
@@ -740,22 +746,16 @@ def update_host_smart_inventory_memberships():
 
 @task(queue=get_local_queuename)
 def migrate_legacy_event_data(tblname):
-    #
-    # NOTE: this function is not actually in use anymore,
-    # but has been intentionally kept for historical purposes,
-    # and to serve as an illustration if we ever need to perform
-    # bulk modification/migration of event data in the future.
-    #
     if 'event' not in tblname:
         return
-    with advisory_lock(f'bigint_migration_{tblname}', wait=False) as acquired:
+    with advisory_lock(f'partition_migration_{tblname}', wait=False) as acquired:
         if acquired is False:
             return
         chunk = settings.JOB_EVENT_MIGRATION_CHUNK_SIZE
 
         def _remaining():
             try:
-                cursor.execute(f'SELECT MAX(id) FROM _old_{tblname};')
+                cursor.execute(f'SELECT MAX(id) FROM _unpartitioned_{tblname};')
```

Comment:
what if we just look if there is at least 1 record
```
select id from _unpartitioned_main_jobevent limit 1;
```

Reply:
@jladdjr  each of the MAX(id), INSERT and DELETE queries is causing sequential scan now, so I think the index on id is needed, if you want to use the current way 

(although there should be a way to move the data with DB procedure without index, in one sequential pass of the table, using a DB cursor ￼ , but it's better to stay with working algorithm as @ryanpetrello says ￼ )

------

I'd say `copy original schema to TMP table` -> `change TMP table schema(remove the primary key)` -> `create partition parent from TMP table schema`  is probably the simplest (one line change, or 2 lines, deleting also the TMP table :slightly_smiling_face: )

So keeping the primary key on the original events table and reusing the algorithm sounds the simplest :-)

Reply:
> I'd say copy original schema to TMP table -> change TMP table schema(remove the primary key) -> create partition parent from TMP table schema is probably the simplest (one line change, or 2 lines, deleting also the TMP table 🙂 )

makes sense - I'm writing this up now and will kick the tires a bit. if things look good we can try to patch things in the scale testing context

Reply:
https://github.com/jladdjr/awx/commit/91b08176884690ef6f3ec69a3cfada884f597571 creates a tmp table that we modify / use as a basis for the partitioned table's schema

Reply:
I made a small tweak to that ^, but after that was able to confirm that the migration went through

Reply:
Going to try to run through a migration w/ a larger set of data and see if we hit any performance issues at a bigger scale (before we go back and try this at true scale)